device/tt_silicon_driver.cpp

```diff
@@ -531,6 +537,28 @@ void TTDevice::do_open() {
 
         this->system_reg_start_offset = (512 - 16) * 1024*1024;
         this->system_reg_offset_adjust = (512 - 32) * 1024*1024;
+    } else if(is_blackhole(device_info.out)) {
+
+        if (bar2_uc_mapping.mapping_id != TENSTORRENT_MAPPING_RESOURCE2_UC) {
+            throw std::runtime_error(std::string("Device ") + std::to_string(index) + " has no BAR4 UC mapping.");
+        }
+
+        // WC mapping
```

Comment:
Why UC rather than WC?

Reply:
I was not sure which region should be used so left both, I have investigated how we should do it and it should be in WC region, changed that